app/scripts/migrations/099.ts

```diff
@@ -0,0 +1,56 @@
+import { cloneDeep, isEmpty } from 'lodash';
+import { v4 as uuid } from 'uuid';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 99;
+
+/**
+ * The core TransactionController uses strings for transaction IDs, specifically UUIDs generated by the uuid package.
+ * For the sake of standardisation and minimising code maintenance, the use of UUIDs is preferred.
+ * This migration updates the transaction IDs to UUIDs.
+ *
+ * @param originalVersionedData
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
```

Comment:
This covers the primary migration, but are we certain there are no other persisted transaction IDs in the rest of the state used as references?

I'm not sure what the easiest way to do this would be, maybe just analysis of some state logs?

Reply:
I've already took a snapshot and see redux state with two transaction. To make sure, I will add that to the comments for you to see as well